test/common/test_point_type_construction.cpp

```diff
@@ -0,0 +1,789 @@
+{
+  PCL_CONSTEXPR const pcl::PrincipalRadiiRSD pt(1.0f, 2.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.r_min, 1.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.r_max, 2.0f);
+}
+
+TEST (PointTypeConstruction, BoundaryDefaultConstruction)
+{
+  PCL_CONSTEXPR const pcl::Boundary pt;
+  PCL_EXPECT_INT_EQ(pt.boundary_point, std::uint8_t{});
+}
+
+TEST (PointTypeConstruction, BoundaryOneScalarConstruction)
+{
+  PCL_CONSTEXPR const pcl::Boundary pt(1u);
+  PCL_EXPECT_INT_EQ(pt.boundary_point, std::uint8_t{1u});
+}
+
+TEST (PointTypeConstruction, PrincipalCurvaturesDefaultConstruction)
+{
+  PCL_CONSTEXPR const pcl::PrincipalCurvatures pt;
+  PCL_EXPECT_FLOAT_EQ(pt.principal_curvature_x, 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.principal_curvature_y, 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.principal_curvature_z, 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.pc1, 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.pc2, 0.0f);
+}
+
+TEST (PointTypeConstruction, PrincipalCurvaturesFiveScalarsConstruction)
+{
+  PCL_CONSTEXPR const pcl::PrincipalCurvatures pt(2.0f, 3.0f, 4.0f, 5.0f, 6.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.principal_curvature_x, 2.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.principal_curvature_y, 3.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.principal_curvature_z, 4.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.pc1, 5.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.pc2, 6.0f);
+}
+
+TEST (PointTypeConstruction, ReferenceFrameDefaultConstruction)
+{
+  PCL_CONSTEXPR const pcl::ReferenceFrame pt;
+  PCL_EXPECT_FLOAT_EQ(pt.rf[0], 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[1], 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[2], 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[3], 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[4], 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[5], 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[6], 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[7], 0.0f);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[8], 0.0f);
+}
+
+TEST (PointTypeConstruction, ReferenceFrameArrayOfScalarsConstruction)
+{
+  PCL_CONSTEXPR const float values[9]{ 1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f, 9.0f };
+  PCL_CONSTEXPR const pcl::ReferenceFrame pt(values);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[0], values[0]);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[1], values[1]);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[2], values[2]);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[3], values[3]);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[4], values[4]);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[5], values[5]);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[6], values[6]);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[7], values[7]);
+  PCL_EXPECT_FLOAT_EQ(pt.rf[8], values[8]);
+}
+
+
+namespace pcl
+{
+
+// to be replaced with std:: when C++20 is available
+// implementations taken from cppreference.com
+
+template<class InputIt, class UnaryPredicate>
+PCL_CONSTEXPR InputIt find_if_not(InputIt first, InputIt last, UnaryPredicate q)
+{
+    for (; first != last; ++first) {
+        if (!q(*first)) {
+            return first;
+        }
+    }
+    return last;
+}
+
+template <class InputIt, class UnaryPredicate>
+PCL_CONSTEXPR bool all_of(InputIt first, InputIt last, UnaryPredicate p)
+{
+    return pcl::find_if_not(first, last, p), last;
+}
```

Comment:
I wonder if we should keep these in the test or move them elsewhere?
@mvieth

Reply:
@mvieth hi! is there anything blocking this issue to be resolved? this PR is almost at the finish line

Reply:
As long as they are only needed here, I would keep them here